lib/Model/Type.cpp

```diff
@@ -1056,20 +1061,54 @@ bool NamedTypedRegister::verify(bool Assert) const {
 }
 
 RecursiveCoroutine<bool> NamedTypedRegister::verify(VerifyHelper &VH) const {
-  const TypedRegister &TR = *this;
-  rc_return VH.maybeFail(CustomName.verify(VH) and rc_recur TR.verify(VH));
+  // Endure the name is valid
+  if (not CustomName.verify(VH))
+    rc_return VH.fail();
+
+  // Ensure the type we're pointing to is scalar
+  if (not isScalar(Type))
+    rc_return VH.fail();
+
+  if (Location == Register::Invalid)
+    rc_return VH.fail();
+
+  // Ensure if fits in the corresponding register
+  auto MaybeTypeSize = rc_recur Type.size(VH);
+
+  // Zero-sized types are not allowed
+  if (not MaybeTypeSize)
+    rc_return VH.fail();
+
+  size_t RegisterSize = model::Register::getSize(Location);
+  if (*MaybeTypeSize > RegisterSize)
+    rc_return VH.fail();
+
+  rc_return VH.maybeFail(rc_recur Type.verify(VH));
```

Comment:
Uhm, I'm a bit confused, why this pops up here now?

Reply:
Because now NamedTypedRegister is not a subclass of TypedRegister anymore, so the `verify` method body had to be duplicated. Not sure if that was the question, the other interpretation I have is "why is this defined here", but the `verify` method for `TypedRegister` was already defined in `Type.cpp` (and still is).

Reply:
Do not duplicate, use a common function.